shims/src/peripherals/gpio.rs

```diff
@@ -156,14 +161,35 @@ impl<'a> Gpio<'a> for GpioShim<'a> {
         }
     }
 
-    fn register_interrupt(
-        &mut self,
-        pin: GpioPin,
-        handler: GpioHandler<'a>,
-    ) -> Result<(), GpioMiscError> {
-        self.handlers[pin] = handler;
+    // TODO: decide functionality when no previous flag registered
+    fn register_interrupt_flag(&mut self, pin: GpioPin, flag: &'a AtomicBool) {
+        self.flags[pin] = match self.flags[pin] {
+            None => Some(flag),
+            Some(_) => unreachable!(),
+        }
+    }
```

Comment:
Can you have this function take all of the flags in 1 go?

i.e. `fn register_interrupt_flag(&mut self, flags: GpioPinArr<&'a AtomicBool>`
and then have the `flags` field be `Option<GpioPinArr<&'a AtomicBool>>.

Reply:
Can do, but I'm not sure if it's best to force all GpioPins to be configured even if just one is being used